### ISSUE TITLE
awlsimhw_linuxcnc: Error on launching linuxcnc example

### ISSUE THREAD

**Issue by uncle-yura** [mbuesch/awlsim#17]
An error occurs when a clean instance hal is passed to the descriptor.

`Traceback (most recent call last):
  File "/home/lcnc/awlsim/examples/linuxcnc-demo/../..//awlsim-linuxcnc-hal", line 438, in <module>
    sys.exit(main())
  File "/home/lcnc/awlsim/examples/linuxcnc-demo/../..//awlsim-linuxcnc-hal", line 390, in main
    server.setRunState(server.STATE_RUN)
  File "/home/lcnc/awlsim/awlsim/coreserver/server.py", line 642, in setRunState
    raise e
AttributeError: 'unicode' object has no attribute 'ready'
`

In file "awlsim/awlsim/common/hwmod.py" empty quotes cause an error.

```
159:  	def setParameterValue(self, name, value):
		"""Set the value of a parameter.
		"""
		self.parameters[name] = value or ""
		self.__identHash = None
```

**Comment by mbuesch** [comment]
Thanks for your report.
I agree that this should be handled more gracefully in awlsimhw_linuxcnc instead of just crashing.

However, can you please elaborate more in detail how it happened that a non-True 'value' was passed to setParameterValue("hal", ...)?
I currently don't see how that could have happened.
Thanks.

**Comment by mbuesch** [comment]
Ok I got it now.
The hal object's boolean value is False.
I'm not sure if that changed in LinuxCNC. But I'll fix the check in Awlsim.

**Comment by mbuesch** [comment]
Fixed in Awlsim 0.72.0.
